kernel/aster-nix/src/syscall/utimens.rs

```diff
@@ -1,91 +1,255 @@
 // SPDX-License-Identifier: MPL-2.0
 
+use alloc::sync::Arc;
 use core::time::Duration;
 
-use super::SyscallReturn;
+use super::{constants::MAX_FILENAME_LEN, SyscallReturn};
 use crate::{
-    fs::{file_table::FileDesc, fs_resolver::FsPath},
+    fs::{
+        file_table::FileDesc,
+        fs_resolver::{FsPath, AT_FDCWD},
+        inode_handle::InodeHandle,
+        path::Dentry,
+    },
     prelude::*,
-    syscall::constants::MAX_FILENAME_LEN,
-    time::timespec_t,
+    time::{clocks::RealTimeCoarseClock, timespec_t},
     util::{read_cstring_from_user, read_val_from_user},
 };
 
+/// The 'sys_utimensat' system call sets the access and modification times of a file specified by `pathname`.
+/// The file is specified by a file descriptor `dirfd` instead of a path name.
+/// The times are specified by a timespec structure.
+/// The `flags` argument is a bit mask that can include the following values:
+/// - `AT_SYMLINK_NOFOLLOW`: If set, the file is not dereferenced if it is a symbolic link.
+/// - `AT_EMPTY_PATH`: If set, an empty pathname is allowed.
```

Comment:
I cant see the `AT_EMPTY_PATH` flags in the manual. https://www.man7.org/linux/man-pages/man3/futimens.3.html
```
       The flags field is a bit mask that may be 0, or include the
       following constant, defined in <fcntl.h>:

       AT_SYMLINK_NOFOLLOW
              If pathname specifies a symbolic link, then update the
              timestamps of the link, rather than the file to which it
              refers.
```

Reply:
You are right. `utimensat()` should not include `AT_EMPTY_PATH `. It's strange that Linux actually implements such a logic in https://github.com/torvalds/linux/blob/master/fs/utimes.c#L91